svc/svc.go

```diff
@@ -107,21 +108,40 @@ func NewStandardHandler(opts HandlerOpts) http.Handler {
 	return middleware.BackgroundContext(h)
 }
 
-// RunServer handles the biolerplate of starting an http server and handling
-// signals gracefully.
-func RunServer(h http.Handler, port string, writeTimeout time.Duration) {
-	// Add timeouts to the server
-	srv := &http.Server{
-		WriteTimeout: writeTimeout * time.Second,
-		Addr:         ":" + port,
-		Handler:      h,
+// BuildServerOpt allows users to customize the http.Server used by RunServer.
+type BuildServerOpt func(*http.Server)
+
+// ServerDefaults specifies default server options to use for RunServer.
+var ServerDefaults = func(srv *http.Server) {
+	srv.Addr = ":8080"
+	srv.WriteTimeout = 5 * time.Second
+	srv.ReadHeaderTimeout = 5 * time.Second
+	srv.IdleTimeout = 120 * time.Second
+}
+
+// WithPort sets the port for the server to run on.
+func WithPort(port string) BuildServerOpt {
```

Comment:
shouldn't `port` be an integer argument? I get you want to concatenate it with `":"` later on, but it looks a bit odd to me

Reply:
I'm gonna go with https://stackoverflow.com/questions/47992477/why-is-port-a-string-and-not-an-integer and keep it a string.

* Ports can be service names (TIL)
* Zero value for int for a port is reserved
* url.Port() also returns a string